File: app/src/main/java/com/friendly_machines/frbpdoctor/logger/Logger.java
package com.friendly_machines.frbpdoctor.logger;

import android.widget.TextView;

public class Logger {
    private static TextView logTextView;

    public static void setLogTextView(TextView textView) {
        logTextView = textView;
    }

    public static void log(String message) {
        if (logTextView != null) {
            logTextView.append(message + "\n");
        }
    }
}

File: app/src/main/java/com/friendly_machines/frbpdoctor/logger/Logger.kt
package com.friendly_machines.frbpdoctor.logger

import android.widget.TextView

object Logger {
    private var logTextView: TextView? = null
    fun setLogTextView(textView: TextView?) {
        logTextView = textView
    }

    fun log(message: String) {
        if (logTextView != null) {
            logTextView!!.append("\n$message")
        }
    }
}